### PR TITLE
docs: add site preview via Netlify

### PR DESCRIPTION
**Pull request by tdstein**
Site preview is available on the action summary page: https://github.com/posit-dev/posit-sdk-py/actions/runs/9452439576?pr=214

I initially attempted posting this as a comment to the pull request, but it became far too complicated to update existing comments on subsequent runs.

### CODE REVIEW

File: docs/requirements-site.txt [hunk 1]
```diff
@@ -0,0 +1,2 @@
+
```
Comment (nealrichardson) [inline docs/requirements-site.txt:1]:
```suggestion
```

File: .github/workflows/site.yaml [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+      - run: make deps
+      - run: make dev
+      - run: make docs
+      - uses: actions/setup-node@v4
+      - id: netlify
+        env:
+            NETLIFY_AUTH_TOKEN: ${{ secrets.NETLIFY_AUTH_TOKEN }}
+        run: |
+          preview_url=$(make --silent -C ./docs netlify | tail -n 1 | tr -d '"')
+          echo "# 🚀 Site Preview" >> $GITHUB_STEP_SUMMARY
```
Comment (nealrichardson) [inline .github/workflows/site.yaml:29]: This just prints in the CI logs right?

Comment (tdstein) [reply .github/workflows/site.yaml:29]: Renders in the preview summary: https://github.com/posit-dev/posit-sdk-py/actions/runs/9452439576?pr=214

(bottom of page) \

Comment (nealrichardson) [reply .github/workflows/site.yaml:29]: Oh nice, that's not bad

Comment (tdstein) [reply .github/workflows/site.yaml:29]: I started playing with writing a comment to the PR, but updating existing comments on re-runs ended up taking more effort than it was worth.

**Pull request merged**